docs/docs/data-structures/tree/schema-evolution/types-of-changes.mdx

```diff
@@ -0,0 +1,213 @@
+
+### Adding Optional Fields
+
+Adding optional fields to existing node types is always [backwards compatible](./index.mdx#understanding-compatibility):
+
+```typescript
+// Before
+class Note extends factory.object("Note", {
+    id: factory.string,
+    text: factory.string,
+}) {}
+
+// After - Safe addition
+class Note extends factory.object("Note", {
+    id: factory.string,
+    text: factory.string,
+    color: factory.optional(factory.string), // New optional field
+}) {}
+```
+
+However, older clients that don't know about this new field will not be able to read it.
+To ensure [forwards compatibility](./index.mdx#understanding-compatibility), use [`allowUnknownOptionalFields`](../../../api/fluid-framework/objectnodeschema-interface#allowunknownoptionalfields-property) to handle new optional fields added by newer clients:
+
+```typescript
+class Note extends factory.object("Note", {
+    id: factory.string,
+    text: factory.string,
+    // Existing optional fields
+}, { allowUnknownOptionalFields: true }) {}
+```
+
+### Making Required Fields Optional
+
+Converting a required field to optional is always [backwards compatible](./index.mdx#backwards-compatibility):
+
+```typescript
+// Before
+class Circle extends factory.object("Circle", {
+    center: Point,
+    radius: factory.number, // Required
+}) {}
+
+// After - Safe change
+class Circle extends factory.object("Circle", {
+    center: Point,
+    radius: factory.optional(factory.number), // Now optional
+}) {}
+```
+
+### Adding New Allowed Types
+
+Adding new allowed types to existing fields is [backwards compatible](./index.mdx#backwards-compatibility), but not [forwards compatible](./index.mdx#forwards-compatibility):
+
+```typescript
+// Before
+class Whiteboard extends factory.object("Whiteboard", {
+    elements: factory.array([TextElement]),
+}) {}
+
+// After - Safe addition
+class Whiteboard extends factory.object("Whiteboard", {
+    elements: factory.array([TextElement, ImageElement]), // New type added
+}) {}
+```
+
+To achieve [forwards compatibility](./index.mdx#forwards-compatibility), staged allowed types can be used to [roll out](./index.mdx#staged-rollouts) support for reading before writing, allowing older clients to read data created with new types (TODO add a guide on this process).
+
+## Incompatible Changes
+
+These changes will break compatibility with existing data and require careful migration:
+
+### Removing Fields
+
+Removing fields from node types breaks older clients:
+
+```typescript
+// Breaking: Removing field
+class Note extends factory.object("Note", {
+    id: factory.string,
+    // text: factory.string, // REMOVED - breaks older clients
+}) {}
+```
+
+### Making Optional Fields Required
+
+Making an optional field required breaks documents that don't have that field:
```

Comment:
```suggestion
Making an optional field required breaks documents whose stored schema does not know about that field:
```

To make it crystal clear that it's the _set of all possible documents_, not the data for any particular document, that determines compatibility.